lightning/src/routing/network_graph.rs

```diff
@@ -947,6 +962,13 @@ impl NetworkGraph {
 			},
 		};
 
+		#[allow(unused_mut, unused_assignments)]
+		let mut announcement_received_time = 0;
```

Comment:
Should we use `None` and condition accordingly in `remove_stale_channels_with_time`? Otherwise, won't we always remove a channel that is waiting on both updates?

Reply:
Right, 0 basically disables the check, `None` and a condition would disable the removal. Given we can re-fetch it, I'd rather air on the side of accidental removal in no-std than never removing in no-std. Of course all that said, I'd be astounded if any no-std users ever touched NetworkGraph.